### DISCUSSION TITLE
Max number of turns

### DISCUSSION THREAD

**Discussion by timnevits** [logancyang/obsidian-copilot#363] [General]
Curious as to why the max number of turns included in the context is capped at 30, so 60 messages total. It's the first time I've explicitly seen this setting. Is this something that happens, for example, with OpenAIs own client and I'm just not aware? Does the conversation degrade after this many turns? Thanks!

**Comment by logancyang** [comment]
@LittleHouse75 This is my own setting, not from OpenAI client. The conversation does degrade depending on the model, GPT-4 degrades the slowest, but for local models it can degrade fast.

The main reason at the beginning was to avoid exceeding a short context window. I was thinking about removing this setting and defaulting to a "summary conversation chain" where if a conversation gets longer than the context window, the older turns get summarized into a shorter blob.

**Reply by timnevits** [reply]
@logancyang thanks much for the explanation. I'll be curious to see how AIs evolve in terms of conversation context.
